pylib/spinnaker/yaml_util.py

```diff
@@ -131,6 +131,42 @@ def __resolve_value(self, value, saw, original):
   def replace(self, text):
     return self.__resolve_value(text, [], text)
 
+
+  def __get_flat_keys(self, container):
+    flat_keys = []
+    for key,value in container.items():
+      if isinstance(value, dict):
+        flat_keys.extend([key + '.' + subkey for subkey in self.__get_flat_keys(value)])
+      else:
+        flat_keys.append(key)
+    return flat_keys
+
+
+  @staticmethod
+  def update_yml_source(path, update_dict):
+    """Update the yaml source at the path according to the update dict.
+
+    All the previous bindings not in the update dict remain unchanged.
+    The yaml file at the path is re-written with the new bindings.
+
+    Args:
+      path [string]: Path to a yaml source file.
+      update_dict [dict(string -> dict)]: Nested dictionary corresponding to
```

Comment:
The type is a dict of object values keyed by strings. It isnt necessarily a dict of dicts (and since it is recursive, at some point it should not be!)